### PR TITLE
Cleaning up a bunch of random TODOs

### PR DESCRIPTION
**Pull request by matt-gardner**

### CODE REVIEW

File: allennlp/data/instance.py [hunk 1]
```diff
@@ -45,12 +45,16 @@ def get_padding_lengths(self) -> Dict[str, Dict[str, int]]:
             lengths[field_name] = field.get_padding_lengths()
         return lengths
 
-    def as_array(self, padding_lengths: Dict[str, Dict[str, int]]) -> Dict[str, DataArray]:
+    def as_arrays(self, padding_lengths: Dict[str, Dict[str, int]] = None) -> Dict[str, DataArray]:
```
Comment (DeNeutoy) [inline allennlp/data/instance.py]: Hmm, this doesn't have the same api as for `Fields` though - maybe the methods on `Instance` and `Dataset` should be called `as_array_dict`?

File: allennlp/models/simple_tagger.py [hunk 2]
```diff
@@ -120,11 +120,10 @@ def tag(self, text_field: TextField) -> Dict[str, Any]:
             An array of shape (text_input_length, num_classes), where each row is a
             distribution over classes for a given token in the sentence.
         """
-        text_field.index(self.vocab)
-        padding_lengths = text_field.get_padding_lengths()
-        array_input = text_field.as_array(padding_lengths)
-        model_input = arrays_to_variables(array_input, add_batch_dimension=True)
-        output_dict = self.forward(tokens=model_input)
+        instance = Instance({'tokens': text_field})
```
Comment (DeNeutoy) [inline allennlp/models/simple_tagger.py]: Disagree with this change - with the others, wrapping the fields in `Instance` makes it less verbose - here, we only have a single field, so we gain no advantage by wrapping it with Instance.

Comment (matt-gardner) [inline allennlp/models/simple_tagger.py]: What we get is consistency, and the fact that we don't have to get the text field padding lengths.  But if you feel strongly about it I'll change it back.

File: allennlp/data/instance.py (hunk 1 above)
Comment (matt-gardner) [inline allennlp/data/instance.py]: Sure, sounds reasonable.

File: allennlp/models/simple_tagger.py (hunk 2 above)
Comment (DeNeutoy) [inline allennlp/models/simple_tagger.py]: I guess it's marginally simpler. down to you.

**Pull request merged**